### ISSUE TITLE
Review the case of having a higher resolution for unit_on than flows in the unit capacity constraint

### ISSUE THREAD

**Issue by datejada** [spine-tools/SpineOpt.jl#797]
**Is your feature request related to a problem? Please describe.**
Review the possibility of having a higher resolution for unit_on than flows in the unit capacity constraint. Especially if the resolutions are not multiple of each other.

@g-moralesespana @manuelma, please complement if needed.

**Describe the solution you'd like**
If the unit_on resolution is lower than the flow resolution, all the units_on variables can be fixed beforehand.

**Describe alternatives you've considered**
The current implementation will work, but you can have more constraints that are needed for the units_on variables.

**Additional context**
Related to changes in #786

**Comment by gnawin** [comment]
@datejada should the following be ?
> Describe the solution you'd like
If the unit_on resolution is **higher** than the flow resolution, all the units_on variables can be fixed beforehand. For example, when unit_on is hourly, and unit_flow is two hourly, then unit_on variables should be fixed to 1 inside the unit_flow `timeslice`.

**Comment by datejada** [comment]
> @datejada should the following be ?
> 
> > Describe the solution you'd like
> > If the unit_on resolution is **higher** than the flow resolution, all the units_on variables can be fixed beforehand. For example, when unit_on is hourly, and unit_flow is two hourly, then unit_on variables should be fixed to 1 inside the unit_flow `timeslice`.

Not necessarily; it can be 1 or 0, depending on the flow, so it is not fixing them to a specific value; it is that we know beforehand that both unit_on variables must have the same value `unit_on(t1)=unit_on(t2)` because the flow is every two hours. If the flow is 0 then both will be zero, and if the flow is >0 then both will be 1.

@g-moralesespana Am I missing something else here? or something you want to add/change

**Comment by gnawin** [comment]
> > @datejada should the following be ?
> > > Describe the solution you'd like
> > > If the unit_on resolution is **higher** than the flow resolution, all the units_on variables can be fixed beforehand. For example, when unit_on is hourly, and unit_flow is two hourly, then unit_on variables should be fixed to 1 inside the unit_flow `timeslice`.
> 
> Not necessarily; it can be 1 or 0, depending on the flow, so it is not fixing them to a specific value; it is that we know beforehand that both unit_on variables must have the same value `unit_on(t1)=unit_on(t2)` because the flow is every two hours. If the flow is 0 then both will be zero, and if the flow is >0 then both will be 1.
> 
> @g-moralesespana Am I missing something else here? or something you want to add/change

Thanks @datejada, sure that the fixed value should depend on the flow value. My point was more about: is this the case that we wanted to discuss, i.e., if the unit_on resolution is **higher** than the flow resolution? In the main post, it is 
>if the unit_on resolution is **lower** than the flow resolution

**Comment by gnawin** [comment]
I did an investigation of this issue, below is what I found.

Time resolution: unit -> 3 h, flow -> 5 h.

The constraint is generated every 3 h. It is working as expected that the highest resolution should be used. Great!

Here are the constraints for the first 6h:
for 0 - 3 h: `flow_{0-5h} <= unit_capacity_{0-3h}`
for 3 - 6 h: `2 * flow_{0-5h} + flow_{5-10h} <= 3 * unit_capacity_{3-6h}` 

So, we can see that the second constraint caps the energy for 3 - 6 h. It can happen that `flow_{5-10h}=3 * unit_capacity_{3-6h}` So we also need upper bounds for the power. This can work if they are additionally generated, but maybe only upper bounds for power are enough. 

Any thoughts? @datejada @g-moralesespana @manuelma

**Comment by DillonJ** [comment]
Ideally what you want are tests that do exactly what you have done @gnawin

**Comment by gnawin** [comment]
The follow-up of this would be to discuss a more general method on how to deal with such cases. TNO will present some ideas in the coming project meeting.
